geoh5py/data/colour.py

```diff
@@ -0,0 +1,172 @@
+#  Copyright (c) 2025 Mira Geoscience Ltd.                                     '
+#                                                                              '
+#  This file is part of geoh5py.                                               '
+#                                                                              '
+#  geoh5py is free software: you can redistribute it and/or modify             '
+#  it under the terms of the GNU Lesser General Public License as published by '
+#  the Free Software Foundation, either version 3 of the License, or           '
+#  (at your option) any later version.                                         '
+#                                                                              '
+#  geoh5py is distributed in the hope that it will be useful,                  '
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of              '
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the               '
+#  GNU Lesser General Public License for more details.                         '
+#                                                                              '
+#  You should have received a copy of the GNU Lesser General Public License    '
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.           '
+# ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import numpy as np
+from numpy.lib import recfunctions as rfn
+
+from ..shared.utils import min_max_scaler
+from .data import Data
+from .primitive_type_enum import PrimitiveTypeEnum
+
+
+logger = logging.getLogger(__name__)
+
+
+class Colour(Data):
+    """
+    Base class for the ternary colour map.
+
+    Warning: actually, GA is storing RGBA but ignore the "A" band (full 255).
+    To mimic this behavior, works with RGB only (and drop A if exists).
+    """
+
+    allowed_parent = [
+        "Points",
+        "Curve",
+        "BlockModel",
+        "Octree",
+        "Grid2D",
+    ]
+
+    def __init__(self, *args, **kwargs):
+        self._verify_parents(kwargs.get("parent", None))
+
+        super().__init__(*args, **kwargs)
+
+    @staticmethod
+    def _array_to_rgb(values: np.ndarray) -> np.ndarray:
+        """
+        Convert a 2D numpy array to RGB.
+
+        :param values: 2D numpy array with 3 or 4 bands.
+
+        :return: A structured array with "r", "g", "b" bands
+        """
+        if values.ndim != 2 or values.shape[1] not in (3, 4):
+            raise ValueError("Values must be a 2D numpy array containing RGB bands.")
+
+        if values.dtype != np.uint8:
+            values = min_max_scaler(values, 0, 255, axis=0)
+
+        if values.shape[1] == 3:
+            values = np.column_stack(
+                (values, np.full(values.shape[0], 255, dtype=np.uint8))
+            )
+
+        values = (
+            values[:, :4]
+            .astype(np.uint8)
```

Comment:
The conversion to int is an issue is the array contains nan from the NDV of GA. We should augment the tests to cover the case.

Reply:
Sure. 
I'll remove the nan values by 90 if in rgb, by 0 if in a,  as no data values is  90, 90, 90, 0